### ISSUE TITLE
Issues related to Solo Machine (ICS06)

### ISSUE THREAD

**Issue by josef-widder** [cosmos/ics#496]
Surfaced from Informal Systems IBC Audit of cosmos-sdk hash https://github.com/cosmos/cosmos-sdk/commit/fe8a891f11dcefe708e43d53549beec808f98687.

## Specification

- [ ] there is a [spec](https://github.com/cosmos/cosmos-sdk/tree/master/x/ibc/light-clients/06-solomachine/spec) in the code repository. It has quite insightful preconditions and postconditions for the functions. These could be used to improve the accessibility of ICS006

- The [Verify](https://github.com/cosmos/ics/tree/master/spec/ics-006-solo-machine-client#state-verification-functions) functions have side effects (`clientState.consensusState.sequence++`). This is not the case in the verify functions of ICS002.
  
## Specification and code

- [ ] the naming in the [spec](https://github.com/cosmos/ics/tree/master/spec/ics-006-solo-machine-client#validity-predicate) does not match the naming in the [implementation](https://github.com/cosmos/cosmos-sdk/blob/fe8a891f11dcefe708e43d53549beec808f98687/x/ibc/light-clients/06-solomachine/types/update.go#L17).
     - the specification of this function talks about `assert`, which is understand as generating an error, while the code aborts the transactions which in the specification typically is written as `abortTransactionUnless`.

- [ ] governance [proposal](https://github.com/cosmos/cosmos-sdk/blob/fe8a891f11dcefe708e43d53549beec808f98687/x/ibc/light-clients/06-solomachine/types/proposal_handle.go) not discussed in the specification

**Comment by cwgoes** [comment]
>  The Verify functions have side effects (clientState.consensusState.sequence++). This is not the case in the verify functions of ICS002.

This is true, and intentional. Do you think this is a bug, or unclearly documented, or something?

> governance proposal not discussed in the specification

See response here - https://github.com/cosmos/ics/issues/497#issuecomment-732541353 - we should make a consistent decision either way.

Otherwise I agree.

**Comment by josef-widder** [comment]
> This is true, and intentional. Do you think this is a bug, or unclearly documented, or something?

At this point it was just surprising, and I wanted to point that out. I guess I was under the (wrong) impression that in general it should not have side effects. Perhaps just adding a sentence saying why it is here would help.
